src/Microsoft.Data.SqlClient/tests/ManualTests/TracingTests/EventCounterTest.cs

```diff
@@ -154,17 +154,24 @@ public void EventCounter_ReclaimedConnectionsCounter_Functional()
 
             long rc = SqlClientEventSourceProps.ReclaimedConnections;
 
-            InternalConnectionWrapper internalConnection = CreateEmancipatedConnection(stringBuilder.ToString());
-
-            GC.Collect();
+            int gcNumber = GC.GetGeneration(CreateEmancipatedConnection(stringBuilder.ToString()));
+            // Specifying the generation number makes it to run faster by avoiding a full GC process
+            GC.Collect(gcNumber);
             GC.WaitForPendingFinalizers();
 
             using (SqlConnection conn = new SqlConnection(stringBuilder.ToString()))
             {
                 conn.Open();
 
-                // when calling open, the connection is reclaimed. 
-                Assert.Equal(rc + 1, SqlClientEventSourceProps.ReclaimedConnections);
+                // when calling open, the connection could reclaimed.
+                if (GC.GetGeneration(conn) == gcNumber)
+                {
+                    Assert.Equal(rc + 1, SqlClientEventSourceProps.ReclaimedConnections);
+                }
+                else
+                {
+                    Assert.Equal(rc, SqlClientEventSourceProps.ReclaimedConnections);
+                }
```

Comment:
How could this be possible? Is it possible that GC runs later and generation number is not changed yet?